### PR TITLE
refactor: Improve Cisco ASA Polling Performance

### PR DESCRIPTION
**Pull request by Rosiak**
- I went through the modules and disabled the ones not supported by the ASA platform.
- I disabled sensor discovery for context based devices as one usually poll the admin context for that purpose, also this output is buggy for contexts on some firmware versions.

Tested on various ASA platforms and on various firmware levels.
This should yield quite a performance improvement on some installs.

asa

### CODE REVIEW

File: includes/discovery/sensors/entity-sensor.inc.php [hunk 1]
```diff
@@ -107,7 +107,7 @@
             }
 
             // echo($descr . "|" . $index . "|" .$current . "|" . $multiplier . "|" . $divisor ."|" . $entry['entPhySensorScale'] . "|" . $entry['entPhySensorPrecision'] . "\n");
-            if ($current == '-127') {
+            if ($current == '-127' || $device['os'] == 'asa' && str_contains($sysDescr, 'sc')) {
```
Comment (laf) [inline includes/discovery/sensors/entity-sensor.inc.php]: Should that be `(device['os'] == 'asa' && str_contains($sysDescr, 'sc'))`?

File: includes/definitions.inc.php [hunk 2]
```diff
@@ -520,6 +520,26 @@
 $config['os'][$os]['over'][0]['text']  = 'Device Traffic';
 $config['os'][$os]['over'][1]['graph'] = 'device_processor';
 $config['os'][$os]['over'][1]['text']  = 'CPU Usage';
+$config['os'][$os]['poller_modules']['aruba-controller']       = 0;
+$config['os'][$os]['poller_modules']['bgp-peers']              = 0;
```
Comment (laf) [inline includes/definitions.inc.php]: The asa can run bgp, I'm not sure if they respond back using our current bgp code but if it does (or you aren't sure) then this should be removed.

Comment (Rosiak) [inline includes/definitions.inc.php]: I considered that too, but didn't have any running BGP so I couldn't test, however looked at ftp://ftp.cisco.com/pub/mibs/supportlists/asa/asa-supportlist.html and can't see that polling BGP info should be supported(Same goes with OSPF).

File: includes/definitions.inc.php [hunk 3]
```diff
@@ -520,6 +520,26 @@
...
+$config['os'][$os]['poller_modules']['cisco-ace-serverfarms']  = 0;
+$config['os'][$os]['poller_modules']['cisco-ace-loadbalancer'] = 0;
+$config['os'][$os]['poller_modules']['cisco-cbqos']            = 0;
+$config['os'][$os]['poller_modules']['cisco-cef']              = 0;
+$config['os'][$os]['poller_modules']['cisco-mac-accounting']   = 0;
+$config['os'][$os]['poller_modules']['cisco-ntp']              = 0;
```
Comment (laf) [inline includes/definitions.inc.php]: This isn't a poller module.

File: includes/definitions.inc.php [hunk 4]
```diff
@@ -520,6 +520,26 @@
...
+$config['os'][$os]['poller_modules']['cisco-voice']            = 0;
+$config['os'][$os]['poller_modules']['hr-mib']                 = 0;
+$config['os'][$os]['poller_modules']['ipmi']                   = 0;
+$config['os'][$os]['poller_modules']['ipSystemStats']          = 0;
+$config['os'][$os]['poller_modules']['junose-atm-vp']          = 0;
+$config['os'][$os]['poller_modules']['netscaler-vsvr']         = 0;
+$config['os'][$os]['poller_modules']['ntp']                    = 0;
+$config['os'][$os]['poller_modules']['ospf']                   = 0;
```
Comment (laf) [inline includes/definitions.inc.php]: I believe the asa's can run ospf as well, again does our code for this not pick it up?

Comment (Rosiak) [inline includes/definitions.inc.php]: Tested against a few running OSPF, nothing got picked up.
Also no OSPF related modules listed under:
ftp://ftp.cisco.com/pub/mibs/supportlists/asa/asa-supportlist.html

File: includes/discovery/sensors/entity-sensor.inc.php (hunk 1 above)
Comment (laf) [inline includes/discovery/sensors/entity-sensor.inc.php]: I think it should be updated to be sure.

File: includes/definitions.inc.php (hunk 3 above)
Comment (Rosiak) [inline includes/definitions.inc.php]: Nice catch!

**Pull request merged**